openslides/assignments/static/js/assignments/base.js

```diff
@@ -183,6 +191,25 @@ angular.module('OpenSlidesApp.assignments', [])
                 // subtitle of search result
                 getSearchResultSubtitle: function () {
                     return "Election";
+                },
+                // override project function of jsDataModel factory
+                project: function (poll_id) {
+                    return $http.post(
+                        '/rest/core/projector/1/prune_elements/',
+                        [{name: 'assignments/assignment', id: this.id, poll: poll_id}]
+                    );
+                },
+                // override isProjected function of jsDataModel factory
+                isProjected: function (poll_id) {
+                    // Returns true if there is a projector element with the name
+                    // 'assignments/assignment'.
+                    var projector = Projector.get(1);
+                    if (typeof projector === 'undefined') return false;
```

Comment:
Both functions are really ugly. There is no self and the `return` inside the code makes it really hard to understand. If we implement this here for consistency we will have much more work in the future.
But the solution is easy. The general function looks like this:
```javascript
function () {
    // Returns true if there is a projector element with the same
    // name and the same id.
    var projector = Projector.get(1);
    if (typeof projector !== 'undefined') {
        var predicate = function (element) {
            return element.name == self.getResourceName() &&
                   typeof element.id !== 'undefined' &&
                   element.id == self.id;
        };
        var isProjected = typeof _.findKey(projector.elements, predicate) === 'string';
    } else {
        var isProjected = false;
    }
    return isProjected
};
```
Don't forget to adapt the comment.